java/common/exception/ErrorMessage.java

```diff
@@ -58,17 +58,20 @@
     public static final ErrorMessage INVALID_RULE_WHEN_CONTAINS_DISJUNCTION =
             new ErrorMessage(19, "Rule '%s' 'when' contains a disjunction.");
     public static final ErrorMessage INVALID_RULE_THEN =
-            new ErrorMessage(20, "Rule '%s' 'then' must strictly be either an attribute ownership, or a relation: '%s'.");
+            new ErrorMessage(20, "Rule '%s' 'then' must strictly be a single of either an attribute ownership, or a relation");
+    public static final ErrorMessage INVALID_RULE_THEN_HAS =
+            new ErrorMessage(21, "Rule '%s' 'then' is trying to assign both an attribute type and a variable attribute value: '%s'." +
```

Comment:
can we try our best to keep this to one line?

Reply:
I can but it makes the line super long.